src/include/stir/recon_buildblock/ProjMatrixByBin.inl

```diff
@@ -104,11 +119,85 @@ get_proj_matrix_elems_for_one_bin(
 #endif
         cache_proj_matrix_elems_for_one_bin(probabilities);
       }
-      symm_ptr->transform_proj_matrix_elems_for_one_bin(probabilities);
+      if ( proj_data_info_sptr->is_tof_data() &&
+                                   this->tof_enabled)
+      {
+          LORInAxialAndNoArcCorrSinogramCoordinates<float> lor;
+          proj_data_info_sptr->get_LOR(lor, bin);
+          LORAs2Points<float> lor2(lor);
+          probabilities.set_bin(bin);
+          // now apply TOF kernel and transform to original bin
+          apply_tof_kernel_and_symm_transformation(probabilities, lor2.p1(), lor2.p2(), symm_ptr);
+
+      }
+      else
+      {
+          // now transform to original bin
+          symm_ptr->transform_proj_matrix_elems_for_one_bin(probabilities);
+      }
       cache_proj_matrix_elems_for_one_bin(probabilities);      
     }
   }  
   // stop_timers(); TODO, can't do this in a const member
 }
 
+void
+ProjMatrixByBin::apply_tof_kernel_and_symm_transformation(ProjMatrixElemsForOneBin& probabilities,
+                                  const CartesianCoordinate3D<float>& point1,
+                                  const CartesianCoordinate3D<float>& point2,
+                                  const unique_ptr<SymmetryOperation>& symm_ptr)  STIR_MUTABLE_CONST
+{
+
+    CartesianCoordinate3D<float> voxel_center;
+    float new_value = 0.f;
+    float low_dist = 0.f;
+    float high_dist = 0.f;
+
+    // The direction can be from 1 -> 2 depending on the bin sign.
+    const CartesianCoordinate3D<float> middle = (point1 + point2)*0.5f;
+    const CartesianCoordinate3D<float> diff = point2 - middle;
+
+    const float lor_length = 1.f / (std::sqrt(diff.x() * diff.x() +
```

Comment:
@KrisThielemans  The length should not change with the detector positions, but stay constant, depending on the size of the coincidence window.

Reply:
The 1/length is to avoid divisions in the for loop. Maybe I should add a comment about it. 

const float lor_length = 1/tof_delta_time_to_mm(-coinc_window/2).

Reply:
I can agree with you that your formula gives (1 over) the lor length. However, for me, the name `lor_length` is misleading. The calculation below is
```
d2 = -inner_product(x, diff) * lor_length;
```
which with current formula becomes
```
d2 = -inner_product(x, diff) / norm(diff);
```
or equivalently
```
unit_vector_along_lor = diff/norm(diff);
d2 = -inner_product(x, unit_vector_along_lor );
```
This seems correct to me: `d2` is (up to the sign) the distance between the projection and the middle of the LOR.
```